src/GraphQl/Resolver/Factory/ItemMutationResolverFactory.php

```diff
@@ -98,11 +102,33 @@ public function __invoke(string $resourceClass = null, string $rootClass = null,
             }
 
             switch ($operationName) {
-                case 'create':
-                case 'update':
+                case 'delete':
+                    if ($item) {
+                        $this->dataPersister->remove($item);
+                        $data['id'] = $args['input']['id'];
+                    } else {
+                        $data['id'] = null;
+                    }
+                    break;
+                default:
                     $context = null === $item ? ['resource_class' => $resourceClass] : ['resource_class' => $resourceClass, 'object_to_populate' => $item];
                     $context += $resourceMetadata->getGraphqlAttribute($operationName, 'denormalization_context', [], true);
                     $item = $this->normalizer->denormalize($args['input'], $resourceClass, ItemNormalizer::FORMAT, $context);
+
+                    if ('create' !== $operationName && 'update' !== $operationName) {
+                        if (null === $this->mutationLocator) {
```

Comment:
I don't really like to throw an exception here because the service is null. I have added my suggestion below.